frappe/website/doctype/personal_data_delete_request/personal_data_delete_request.js

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2019, Frappe Technologies and contributors
+// For license information, please see license.txt
+
+frappe.ui.form.on('Personal Data Delete Request', {
+	refresh: function(frm) {
+		if(frappe.user.has_role('System Manager') && frm.doc.status == 'Pending Approval'){
```

Comment:
role validations should also be server side

Reply:
added validations on the server side.